packages/debug/src/browser/debug-frontend-application-contribution.ts

```diff
@@ -640,10 +640,10 @@ export class DebugFrontendApplicationContribution extends AbstractViewContributi
     registerCommands(registry: CommandRegistry): void {
         super.registerCommands(registry);
         registry.registerCommand(DebugCommands.START, {
-            execute: (config?: DebugSessionOptions) => this.start(false, config)
+            execute: (config?: DebugSessionOptions) => this.start(false, isAnchor(config) ? undefined : config)
         });
         registry.registerCommand(DebugCommands.START_NO_DEBUG, {
-            execute: (config?: DebugSessionOptions) => this.start(true, config)
+            execute: (config?: DebugSessionOptions) => this.start(true, isAnchor(config) ? undefined : config)
```

Comment:
Sorry to change my mind here, but even though the cause of the problem is that an Anchor is being passed in, but what we really want to guarantee is that the `config` is an instance of `DebugSessionOptions` - the problem with the `CommandHandler` system is that we're allowed to say `config?: DebugSessionOption`, but in fact anything can be passed in - Anchors, or anything else..

So rather than adding a check for whether this is an Anchor, it would be more correct in this context to implement a check that something is a DebugSessionOptions. Something like.

```typescript
export namespace DebugSessionOptions {
    export function is(candidate: unknown): candidate is DebugSessionOptions {
        const options = candidate as DebugSessionOptions;
        return typeof options === 'object' && !!candidate && 'configuration' in candidate && DebugConfiguration.is(options.candidate);
    }
}
```
Again, sorry for the change of direction!

Reply:
I would say hold off on this pending a general resolution of the problems introduced by `window.menuBarVisibility` `compact`. As I say [here](https://github.com/eclipse-theia/theia/issues/10571#issuecomment-999769286), the correct, non-breaking solution is probably just to fix the behavior of the menu introduced by that PR, rather than fix the commands one by one or change the existing behavior of real context menus.